vyxal/helpers.py

```diff
@@ -96,19 +96,24 @@ def digits(num: NUMBER_TYPE) -> List[int]:
 
 
 @lazylist
-def enumerate_md(haystack: VyList, index_stack: List[int] = []) -> VyList:
```

Comment:
Extracted this into a helper because Deepsource was complaining that `[]` as a default argument was dangerous because it's mutable. We're not mutating it so we could've just left that in, but it's one less thing Deepsource will complain about and I don't want to allow ourselves to accidentally set `index_stack` to something other than `[]` when calling `enumerate_md`.

Reply:
Why not just use a tuple?

Reply:
I didn't like the idea of exposing implementation details by allowing callers to set `index_stack` themselves, but if an optional argument is more idiomatic, we can go with that, sure

Reply:
I think the idiomatic way to specify that an argument is an implementation detail and shouldn't be used by API consumers, is just to name the argument with a leading underscore:

` def enumerate_md(haystack: VyList, _index_stack: List[int] = ()) -> VyList:`